Yosevu/css 120 migrate link to core tokens

yosevu opened the pull request:
<!-- Summarize your changes in the Title field -->

## Description
- Update windows high contrast mode
- Update YAML examples
- Remove t-shirt sizes
- Add Static White and Static Black variants

## How and where has this been tested?
- Tested locally referencing the Progress-bar XD file.
- Tested WHCM with forced colors in Chrome Emulation

## Screenshots
<!-- If applicable, add screenshots to show what you changed -->


## To-do list
<!-- Put an "x" to indicate you've done each of the following -->
- [x] If my change impacts other components, I have tested to make sure they don't break.
- [x] If my change impacts documentation, I have updated the documentation accordingly.
- [x] I have read the [CONTRIBUTING document](/.github/CONTRIBUTING.md).
- [x] I have tested these changes in Windows High Contrast mode.
<!-- If this pull request isn't ready, add any remaining tasks here -->
- [ ] This pull request is ready to merge.

lunarfusion commented on components/link/index.css:
```diff
@@ -10,55 +10,95 @@ OF ANY KIND, either express or implied. See the License for the specific languag
   -webkit-text-decoration-skip: objects;
   text-decoration: underline;
-  transition: color var(--spectrum-global-animation-duration-100) ease-in-out;
+  transition: color var(--spectrum-animation-duration-100) ease-in-out;
```
Should this be established as a custom property up top?

yosevu replied:
That makes sense. Updated.

lunarfusion commented on components/link/index.css:
```diff
@@ -10,55 +10,95 @@ OF ANY KIND, either express or implied. See the License for the specific languag
     text-decoration-style: double;
   }
+
+  color: var(--spectrum-link-text-color-primary-default);
```
Missing -mod on custom properties.

lunarfusion commented on components/link/index.css:
```diff
@@ -10,55 +10,95 @@ OF ANY KIND, either express or implied. See the License for the specific languag
+/* Windows high contrast mode */
+@media (forced-colors: active) {
+  &--staticBlack {
+    --highcontrast-link-text-color-black: ButtonFace;
```
I think this should be ButtonText instead of ButtonFace. Buttonface does not seem to be a text color. Here is our demo that was tested on a PC in dark mode:

![WHCM-dark-pc](https://user-images.githubusercontent.com/25614178/185699391-e09e894e-a195-416c-b5ba-1b550a925a04.png)

yosevu replied:
Yeah, these were the values already on the branch. I forgot to reference that demo.

lunarfusion commented on components/link/index.css:
```diff
@@ -10,55 +10,95 @@ OF ANY KIND, either express or implied. See the License for the specific languag
+    color: var(--spectrum-link-text-color-primary-active);
+  }
+
+  &:focus-visible {
```
Focus-visible looks like a repeat, also appears on line 68. Can these be combined?

yosevu replied:
Ah thanks. Updated.

lunarfusion commented on components/link/index.css:
```diff
@@ -10,55 +10,94 @@ OF ANY KIND, either express or implied. See the License for the specific languag
     text-decoration-style: double;
   }
+
+  &:hover {
```
Just a thought - sorry I didn't think of this before. In the variations after this, your states are in the order of 

1. hover
2. active 
3. focus 

Do you think it would be super extra tidy if the main component were in that order too (hover - active - focus-visible)?

yosevu replied:
Oh good call. The order of actions is important - Lord Vader Hates Fluffy Animals:

1. link
2. visited
3. hover
4. focus
5. active

lunarfusion replied:
gotta love a subversive acronym!

bernhard-adobe reviewed (commented):
Are we missing a bunch of `-mod`color mods here for custom colors?

bernhard-adobe commented:
VRT is: https://spectrum-visual-regression.ci.corp.adobe.com/view/Spectrum%20CSS/job/css-vrt-test/40/
[1504 link.zip](https://github.com/adobe/spectrum-css/files/9410788/1504.link.zip)

bernhard-adobe commented:
VRT is: https://spectrum-visual-regression.ci.corp.adobe.com/view/Spectrum%20CSS/job/css-vrt-test/40
[1504 link.zip](https://github.com/adobe/spectrum-css/files/9410824/1504.link.zip)

yosevu commented:
I added missing mod tokens and updated whcm styles
<img width="237" alt="Screen Shot 2022-08-24 at 6 12 46 AM" src="https://user-images.githubusercontent.com/16216104/186273439-e7fe71c9-50d8-4168-aeb9-1d7083f09b73.png">
<img width="232" alt="Screen Shot 2022-08-24 at 6 12 31 AM" src="https://user-images.githubusercontent.com/16216104/186273443-5a673eb9-3f4f-42dd-8357-3a664bf9afb0.png">

pfulton commented:
Released beta:
`4.0.0-beta.0`

@yosevu can you open an [PR against the SWC repo](https://github.com/adobe/spectrum-web-components/pulls) with this update?
